jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/servlets/ActionPrefixesRW.java

```diff
@@ -0,0 +1,141 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jena.fuseki.servlets;
+
+import org.apache.jena.atlas.logging.FmtLog;
+import org.apache.jena.riot.WebContent;
+import org.apache.jena.riot.web.HttpNames;
+
+import org.apache.jena.fuseki.servlets.prefixes.PrefixUtils;
+
+import java.io.IOException;
+
+public class ActionPrefixesRW extends ActionPrefixesR {
+
+    public ActionPrefixesRW() {}
+
+    @Override
+    protected void doOptions(HttpAction action) {
+        ActionLib.setCommonHeadersForOptions(action);
+        action.setResponseHeader(HttpNames.hAllow, "GET,OPTIONS,POST");
+        ServletOps.success(action);
+    }
+
+    public void validateDelete(HttpAction action) {
+        validate(action);
+        String prefixToRemove = action.getRequestParameter("removeprefix");
+        if (prefixToRemove == null || prefixToRemove.isEmpty() || !PrefixUtils.prefixIsValid(prefixToRemove)) {
+            ServletOps.errorBadRequest("Remove operation unsuccessful!");
+            return;
+        }
+    }
+
+    @Override
+    protected void doDelete(HttpAction action) {
+        validateDelete(action);
+        action.beginWrite();
+
+        try {
+            String prefixToRemove = action.getRequestParameter("removeprefix");
+            prefixes(action).removePrefix(prefixToRemove);
+            FmtLog.info(action.log, "[%d] Remove %s:", action.id, prefixToRemove);
+
+            action.commit();
+
+            // Indicate success
+            ServletOps.success(action);
+
+            // Build the response.
+            action.setResponseContentType(WebContent.contentTypeJSON);
+            action.getResponseOutputStream().print("");
+
+        } catch (RuntimeException | IOException ex) {
+            try { action.abort(); }
+            catch (Throwable th ) {
+                FmtLog.warn(action.log, th, "[%d] POST prefix = %s", action.id);
+            }
+            ServletOps.errorOccurred(ex);
+        } finally {
+            action.end();
+        }
+    }
+
+
+    public void validatePost(HttpAction action) {
+        validate(action);
+        String prefix = action.getRequestParameter("prefix");
+        String uri = action.getRequestParameter("uri");
+        String prefixToRemove = action.getRequestParameter("removeprefix");
```

Comment:
So I guess this is why `removeprefix` is a separate parameter because we have `POST` allowing for both update and delete, might be cleaner to just limit `POST` to updating a prefix and keep `DELETE` for deleting a prefix